src/app/beer_garden/router.py

```diff
@@ -40,6 +41,34 @@
 gardens: Dict[str, Garden] = {}
 gardens_lock = threading.Lock()
 
+
+def route_garden_sync(garden_name: str = None):
+    # If a Garden Name is provided, determine where to route the request
+    if garden_name:
+        if garden_name == config.get("garden.name"):
+            beer_garden.garden.sync_garden()
+        else:
+            forward(
+                Operation(operation_type="GARDEN_SYNC", target_garden_name=garden_name)
+            )
+
+    else:
+        # Iterate over all gardens and forward the sync request
+        with gardens_lock:
+            for garden in gardens.values():
+                if garden.name == config.get("garden.name"):
+                    beer_garden.garden.sync_garden()
+                else:
+                    forward(
+                        Operation(
+                            operation_type="GARDEN_SYNC", target_garden_name=garden.name
+                        )
+                    )
+
+        # Final sync runs just in case this doesn't have any children
+        beer_garden.garden.sync_garden()
```

Comment:
Won't this run the `sync_garden` (and so publish the event) twice? Should it just be:

```python
                if garden.name != config.get("garden.name"):
                    forward(
                        Operation(
                            operation_type="GARDEN_SYNC", target_garden_name=garden.name
                        )
                    )
```

and then run the last sync_garden?